backend/pkg/api/instances_test.go

```diff
@@ -294,3 +295,102 @@ func TestGetInstanceStatusHistory(t *testing.T) {
 	assert.Equal(t, history[3].Status, InstanceStatusUpdateGranted)
 	assert.Equal(t, history[3].Version, "1.0.1")
 }
+
+func TestUpdateInstanceFact(t *testing.T) {
+	a := newForTest(t)
+	defer a.Close()
+
+	start := time.Now()
+
+	tTeam, err := a.AddTeam(&Team{Name: "test_team"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	tApp, err := a.AddApp(&Application{Name: "test_app", TeamID: tTeam.ID})
+	if err != nil {
+		t.Fatal(err)
+	}
+	tPkg, err := a.AddPackage(&Package{Type: PkgTypeOther, URL: "http://sample.url/pkg", Version: "12.1.0", ApplicationID: tApp.ID, Arch: ArchAMD64})
+	if err != nil {
+		t.Fatal(err)
+	}
+	tChannel, err := a.AddChannel(&Channel{Name: "test_channel", Color: "blue", ApplicationID: tApp.ID, PackageID: null.StringFrom(tPkg.ID), Arch: ArchAMD64})
+	if err != nil {
+		t.Fatal(err)
+	}
+	tGroup, err := a.AddGroup(&Group{Name: "group1", ApplicationID: tApp.ID, ChannelID: null.StringFrom(tChannel.ID), PolicyUpdatesEnabled: true, PolicySafeMode: false, PolicyPeriodInterval: "15 minutes", PolicyMaxUpdatesPerPeriod: 2, PolicyUpdateTimeout: "60 minutes"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	tInstance1, err := a.RegisterInstance(uuid.New().String(), "", "10.0.0.1", "1.0.0", tApp.ID, tGroup.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	tInstance2, err := a.RegisterInstance(uuid.New().String(), "", "10.0.0.2", "1.0.0", tApp.ID, tGroup.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = a.RegisterInstance(uuid.New().String(), "", "10.0.0.3", "1.0.1", tApp.ID, tGroup.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// check tInstance1 in twice
+	_, err = a.GetUpdatePackage(tInstance1.ID, "", "10.0.0.1", "1.0.0", tApp.ID, tGroup.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// switch tInstance2 version
+	_, err = a.GetUpdatePackage(tInstance2.ID, "", "10.0.0.2", "1.0.1", tApp.ID, tGroup.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ts := time.Now()
+	elapsed := ts.Sub(start)
+
+	update := a.updateInstanceFact(&ts, &elapsed)
```

Comment:
```suggestion
	err = a.updateInstanceFact(&ts, &elapsed)
	if err != nil {
		t.Fatal(err)
	}
```
same below at the other call site